relay-server/src/utils/rate_limits.rs

```diff
@@ -372,18 +387,53 @@ where
         let mut enforcement = Enforcement::default();
 
         if let Some(category) = summary.event_category {
-            let event_limits = (self.check)(scoping.item(category), 1)?;
-            let longest = event_limits.longest();
-            enforcement.event = CategoryLimit::new(category, 1, longest);
-
-            // Record the same reason for attachments, if there are any.
-            enforcement.attachments = CategoryLimit::new(
-                DataCategory::Attachment,
-                summary.attachment_quantity,
-                longest,
-            );
-
-            rate_limits.merge(event_limits);
+            if category == DataCategory::Transaction {
+                if summary.transaction_metrics_extracted {
+                    // Check and increment indexing quota.
+                    let index_limits = (self.check)(scoping.item(DataCategory::Transaction), 1)?;
+                    let longest = index_limits.longest();
+                    enforcement.event = CategoryLimit::new(DataCategory::Transaction, 1, longest);
+                    rate_limits.merge(index_limits);
+
+                    // Check processing quota, but DO NOT increment.
+                    let processing_limits =
+                        (self.check)(scoping.item(DataCategory::TransactionProcessed), 0)?;
```

Comment:
I don't understand the logic behind checking the processed quota with a quantity of 0.